proposals/devcontainer-distribution.md

```diff
@@ -0,0 +1,71 @@
+
+## Releases
+
+As a distribution and storage mechanism we propose using GitHub releases. The releases would contain compressed files with the source code of the feature or template. Together with the release the corresponding metadata file would be included either **devcontainer-feature.json**, **devcontainer-template.json** or **devcontainer-collection.json**.
+
+## Collections
+
+Features and/or templates can be distributed both as a single unit or as a **collection** of them. In case they are released as a collection the release should include a **devcontainer-collection.json** file that groups the information of each feature/template contained.
+
+| Property | Type | Description |
+| :--- | :--- | :--- |
+| publisher | string | The organization/user that publishes this group of templates/features. Must match with the repository user in github.|
+| repository | string | The repository where this collection is contained.|
+| version | string | Version information of this collection list.|
+| features | array | The list of features that are contained in this collection. Same information as the metadata file for the feature.|
+| templates | array | The list of templates contained in this collection. Same information as the metadata file for the definition.|
+
+Its important to note the following about collections:
+
+- It is recomended that they are created automatically as part of the release process.
+- Ids would get the collection repository added to them. (e.g. for 'https://github.com/community/features' then 'community/features' to 'myfeature1' for a feature with that id)
+- The links to the release files would be added.
+- Date.
+- Version.
+
+## Templates specifics
+
+By their nature templates are single use since they are just a group of files that are copied over to an end users repository and used to kickstart a **dev container** configuration.
+
+Due to this the implementing application needs to ensure that the user is able to inspect all files included in the template and that no code is executed automatically until a specific action by a user demands it. 
+
+## Features specifics
+
+Since features contain code that can be executed multiple times by different users the security of the contents is important, for this we propose:
+
+- Any application that implements the [Dev container spec](../docs/specs/devcontainer-reference.md) must provide a way for the user to add a feature to their `devcontainer.json` file in a way that the code can be inspected before it executes the first time.
+- This action should create a **devcontainer.lock** file next to the main `devcontainer.json` file that contains the checksums of the particular release referenced.
+- Users can commit this file to their repository so that the application can berify the checksum each time it rebuilds the container, throwing an error if they not match.
```

Comment:
```suggestion
- Users can commit this file to their repository so that the application can berify the checksum each time it rebuilds the container, throwing an error if they do not match.
```